sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/MergingSessionsIterator.scala

```diff
@@ -118,7 +118,11 @@ class MergingSessionsIterator(
       val inputRow = inputIterator.next()
       nextGroupingKey = groupingWithoutSessionProjection(inputRow).copy()
       val session = sessionProjection(inputRow)
-      nextGroupingSession = session.getStruct(0, 2).copy()
+      val groupingSession = session.getStruct(0, 2)
+      if (groupingSession == null) {
```

Comment:
Let's be conservative, `errorOnIterator = true` to invalidate the iterator. Please apply the same to below as well.

Reply:
ah thanks! addressed